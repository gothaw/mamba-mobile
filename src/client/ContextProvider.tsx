import React, { createContext, FunctionComponent } from 'react';

import { ContextInterface } from "../types";
import { theme } from "../theme";

type Props = {
  children: React.ReactNode
};

const Context = createContext<ContextInterface | null>(null);

const ContextProvider: FunctionComponent<Props> = ({children}) => {
  return (
    <Context.Provider value={theme}>
      {children}
    </Context.Provider>
  );
};

export { Context, ContextProvider };
