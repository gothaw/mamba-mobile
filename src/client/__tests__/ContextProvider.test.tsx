import { useContext } from "react";
import { Text } from "react-native";
import { render, waitFor } from "@testing-library/react-native";

import { Context, ContextProvider } from "../ContextProvider";

const TestComponent = () => {
  const context = useContext(Context);

  return <Text testID={"component"} style={{...context.theme.container}}>Component</Text>;
};

const setup = () => {
  return render(
    <ContextProvider>
      <TestComponent/>
    </ContextProvider>
  );
};

describe("ContextProvider", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it("should render", async () => {
    await waitFor(() => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  it("should render TestComponent with correct styling", async () => {
    const testComponent = await wrapper.findByTestId("component");

    await waitFor(() => {
      expect(testComponent).toBeTruthy();
      expect(wrapper.toJSON().props.style).toStrictEqual({flex: 1, position: "relative"});
    });
  });
});