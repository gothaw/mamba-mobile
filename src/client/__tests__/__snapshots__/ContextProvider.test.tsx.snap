// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextProvider should render 1`] = `
<Text
  style={
    {
      "flex": 1,
      "position": "relative",
    }
  }
  testID="component"
>
  Component
</Text>
`;
